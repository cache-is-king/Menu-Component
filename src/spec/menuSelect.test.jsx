import React from 'react';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import MenuSelector from '../client/app/MenuSelector';
import fakeMenus from './fakeData';

Enzyme.configure({ adapter: new Adapter() });

describe('MenuSelector Component', () => {
  let component = '';
  test('should render lunch EntreeMenu correctly', () => {
    component = Enzyme.shallow(<MenuSelector menu={fakeMenus.lunch} type={'lunch'} tag={'none'} />);
    expect(component).toMatchSnapshot();
  });

  test('should render dinner EntreeMenu component correctly', () => {
    component = Enzyme.shallow(<MenuSelector menu={fakeMenus.lunch} type={'dinner'} tag={'none'} />);
    expect(component).toMatchSnapshot();
  });

test('should render dessert Menu component correctly', () => {
    component = Enzyme.shallow(<MenuSelector menu={fakeMenus.dessert} type={'dessert'} tag={'none'} />);
    expect(component).toMatchSnapshot();
  });

  test('should render filtered Menu component correctly', () => {
    component = Enzyme.shallow(<MenuSelector menu={fakeMenus.vegan} type={'dinner'} tag={'vegan'} />);
    expect(component).toMatchSnapshot();
  });
});
