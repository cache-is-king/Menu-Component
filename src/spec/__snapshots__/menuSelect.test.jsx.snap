// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuSelector Component should render dessert Menu component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuType
    menu={
      Array [
        Object {
          "cost": 99,
          "foodItem": "cake",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "pie",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "whipped cream",
          "tags": "none",
        },
      ]
    }
    tag="none"
    type="dessert"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "menu": Array [
        Object {
          "cost": 99,
          "foodItem": "cake",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "pie",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "whipped cream",
          "tags": "none",
        },
      ],
      "tag": "none",
      "type": "dessert",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "menu": Array [
          Object {
            "cost": 99,
            "foodItem": "cake",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "sausages",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "pie",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "whipped cream",
            "tags": "none",
          },
        ],
        "tag": "none",
        "type": "dessert",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MenuSelector Component should render dinner EntreeMenu component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuType
    menu={
      Array [
        Object {
          "cost": 99,
          "foodItem": "sandwich",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "tacos",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "burger",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "chips",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "steak",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "ham",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "turkey",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "pop tarts",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "cheetos",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "snickers",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "apple",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "jicama",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "pineapple",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "paper",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "ramen",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "mochi",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "tuna melt",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "chilli",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "mushrooms",
          "tags": "none",
        },
      ]
    }
    tag="none"
    type="dinner"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "menu": Array [
        Object {
          "cost": 99,
          "foodItem": "sandwich",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "tacos",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "burger",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "chips",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "steak",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "ham",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "turkey",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "pop tarts",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "cheetos",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "snickers",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "apple",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "jicama",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "pineapple",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "paper",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "ramen",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "mochi",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "tuna melt",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "chilli",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "mushrooms",
          "tags": "none",
        },
      ],
      "tag": "none",
      "type": "dinner",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "menu": Array [
          Object {
            "cost": 99,
            "foodItem": "sandwich",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "sausages",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "tacos",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "burger",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "chips",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "steak",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "ham",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "turkey",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "pop tarts",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "cheetos",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "snickers",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "apple",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "jicama",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "pineapple",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "paper",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "ramen",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "mochi",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "tuna melt",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "chilli",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "mushrooms",
            "tags": "none",
          },
        ],
        "tag": "none",
        "type": "dinner",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MenuSelector Component should render filtered Menu component correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuType
    menu={
      Array [
        Object {
          "cost": 99,
          "foodItem": "sandwich",
          "tags": "vegan",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "vegan",
        },
        Object {
          "cost": 35,
          "foodItem": "tacos",
          "tags": "vegan",
        },
        Object {
          "cost": 10,
          "foodItem": "burger",
          "tags": "vegan",
        },
      ]
    }
    tag="vegan"
    type="dinner"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "menu": Array [
        Object {
          "cost": 99,
          "foodItem": "sandwich",
          "tags": "vegan",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "vegan",
        },
        Object {
          "cost": 35,
          "foodItem": "tacos",
          "tags": "vegan",
        },
        Object {
          "cost": 10,
          "foodItem": "burger",
          "tags": "vegan",
        },
      ],
      "tag": "vegan",
      "type": "dinner",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "menu": Array [
          Object {
            "cost": 99,
            "foodItem": "sandwich",
            "tags": "vegan",
          },
          Object {
            "cost": 50,
            "foodItem": "sausages",
            "tags": "vegan",
          },
          Object {
            "cost": 35,
            "foodItem": "tacos",
            "tags": "vegan",
          },
          Object {
            "cost": 10,
            "foodItem": "burger",
            "tags": "vegan",
          },
        ],
        "tag": "vegan",
        "type": "dinner",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`MenuSelector Component should render lunch EntreeMenu correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuType
    menu={
      Array [
        Object {
          "cost": 99,
          "foodItem": "sandwich",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "tacos",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "burger",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "chips",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "steak",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "ham",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "turkey",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "pop tarts",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "cheetos",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "snickers",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "apple",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "jicama",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "pineapple",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "paper",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "ramen",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "mochi",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "tuna melt",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "chilli",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "mushrooms",
          "tags": "none",
        },
      ]
    }
    tag="none"
    type="lunch"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "menu": Array [
        Object {
          "cost": 99,
          "foodItem": "sandwich",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "sausages",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "tacos",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "burger",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "chips",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "steak",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "ham",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "turkey",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "pop tarts",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "cheetos",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "snickers",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "apple",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "jicama",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "pineapple",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "paper",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "ramen",
          "tags": "none",
        },
        Object {
          "cost": 99,
          "foodItem": "mochi",
          "tags": "none",
        },
        Object {
          "cost": 50,
          "foodItem": "tuna melt",
          "tags": "none",
        },
        Object {
          "cost": 35,
          "foodItem": "chilli",
          "tags": "none",
        },
        Object {
          "cost": 10,
          "foodItem": "mushrooms",
          "tags": "none",
        },
      ],
      "tag": "none",
      "type": "lunch",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "menu": Array [
          Object {
            "cost": 99,
            "foodItem": "sandwich",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "sausages",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "tacos",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "burger",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "chips",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "steak",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "ham",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "turkey",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "pop tarts",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "cheetos",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "snickers",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "apple",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "jicama",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "pineapple",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "paper",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "ramen",
            "tags": "none",
          },
          Object {
            "cost": 99,
            "foodItem": "mochi",
            "tags": "none",
          },
          Object {
            "cost": 50,
            "foodItem": "tuna melt",
            "tags": "none",
          },
          Object {
            "cost": 35,
            "foodItem": "chilli",
            "tags": "none",
          },
          Object {
            "cost": 10,
            "foodItem": "mushrooms",
            "tags": "none",
          },
        ],
        "tag": "none",
        "type": "lunch",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
